### PR TITLE
Add MXE build lane.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by vargaz** [comment]
So the build succeeds, but package creation fails because it seems like it cannot package .exe files on non-windows platforms.

**Comment by vargaz** [comment]
So this works now, the only problem is that the directory inside the nuget where the files are is called
tools/osx.10.12-x64-mxe.

**Comment by vargaz** [comment]
Tested it during the android sdk build.

**Comment by vargaz** [comment]
Merge to get test packages. Comments will be addressed in a follow-up PR.

File: llvm.proj [hunk 1]
```diff
@@ -55,6 +55,21 @@
...
+# -DCMAKE_EXE_LINKER_FLAGS=-static is needed so that we don't dynamically link with any of the mingw gcc support libs.
+-->
+	<ItemGroup Condition=" '$(TargetArchitecture)' == 'x64-mxe' ">
+	  <_LLVMBuildArgs Include="-DCMAKE_EXE_LINKER_FLAGS=-static"/>
+	  <_LLVMBuildArgs Include="-DCROSS_TOOLCHAIN_FLAGS_NATIVE=-DCMAKE_TOOLCHAIN_FILE=cmake/modules/NATIVE.cmake"/>
+	  <_LLVMBuildArgs Include="-DCMAKE_TOOLCHAIN_FILE=cmake/modules/mxe-Win64.cmake"/>
+	  <_LLVMBuildArgs Include="-DLLVM_ENABLE_THREADS=Off"/>
+	  <_LLVMBuildArgs Include="-DLLVM_BUILD_EXECUTION_ENGINE=Off"/>
+	  </ItemGroup>
+
```
Comment (akoeplinger) [inline llvm.proj:72]: there's some tabs/spaces mismatch here

**Review by akoeplinger — COMMENTED** [review]
LGTM apart from the tabs/spaces issue.

**Pull request merged**